src/reconcile/rec_write.c

```diff
@@ -1201,138 +1216,116 @@ err:	__wt_scr_free(session, &tmp);
 /*
  * __rec_txn_read --
  *	Return the update in a list that should be written (or NULL if none can
- * be written).
+ *	be written).
  */
 static int
 __rec_txn_read(WT_SESSION_IMPL *session, WT_RECONCILE *r,
     WT_INSERT *ins, void *ripcip, WT_CELL_UNPACK *vpack, WT_UPDATE **updp)
 {
 	WT_BTREE *btree;
-	WT_DECL_TIMESTAMP(max_timestamp)
 	WT_PAGE *page;
-	WT_UPDATE *upd, *upd_list;
-	size_t update_mem;
+	WT_UPDATE *first_ts_upd, *first_txn_upd, *first_upd, *next_upd, *upd;
+	wt_timestamp_t *timestampp;
 	uint64_t max_txn, txnid;
-	bool skipped;
+	bool all_visible, uncommitted;
 
 	*updp = NULL;
 
 	btree = S2BT(session);
 	page = r->page;
+	first_ts_upd = first_txn_upd = NULL;
+	max_txn = WT_TXN_NONE;
+	uncommitted = false;
 
 	/*
 	 * If called with a WT_INSERT item, use its WT_UPDATE list (which must
 	 * exist), otherwise check for an on-page row-store WT_UPDATE list
 	 * (which may not exist). Return immediately if the item has no updates.
 	 */
-	if (ins == NULL) {
-		if ((upd_list = WT_ROW_UPDATE(page, ripcip)) == NULL)
-			return (0);
-	} else
-		upd_list = ins->upd;
+	if (ins != NULL)
+		first_upd = ins->upd;
+	else if ((first_upd = WT_ROW_UPDATE(page, ripcip)) == NULL)
+		return (0);
 
-	skipped = false;
-	update_mem = 0;
-	max_txn = WT_TXN_NONE;
-#ifdef HAVE_TIMESTAMPS
-	__wt_timestamp_set_zero(&max_timestamp);
-#endif
+	for (upd = first_upd; upd != NULL; upd = next_upd) {
+		next_upd = upd->next;
```

Comment:
I think the `next_upd = upd->next ... upd = next_upd` work is a holdover from a previous version where the list was modified? I don't see any need for it at this point.